compiler/passes/buildDefaultFunctions.cpp

```diff
@@ -209,35 +223,44 @@ static FnSymbol* function_exists(const char* name,
 }
 
 
+static void fixup_getter(AggregateType* ct, Symbol *field,
+                         bool fieldIsConst, bool recordLike,
+                         FnSymbol* fn)
+{
+  std::vector<BaseAST*> asts;
+  collect_asts(fn, asts);
+  for_vector(BaseAST, ast, asts) {
+    if (CallExpr* call = toCallExpr(ast)) {
+      if (call->isNamed(field->name) && call->numActuals() == 2) {
+        if (call->get(1)->typeInfo() == dtMethodToken &&
+            call->get(2)->typeInfo() == ct) {
+          Expr* arg2 = call->get(2);
+          call->replace(new CallExpr(PRIM_GET_MEMBER,
+                                     arg2->remove(),
+                                     new_CStringSymbol(field->name)));
+        }
+      }
+    }
+  }
+  fn->addFlag(FLAG_FIELD_ACCESSOR);
+  if (fieldIsConst)
+    fn->addFlag(FLAG_REF_TO_CONST);
+  else if (recordLike)
+    fn->addFlag(FLAG_REF_TO_CONST_WHEN_CONST_THIS);
+}
+
 // Getter and setter functions are provided by the compiler if not supplied by
 // the user.
 // These functions have the same binding strength as if they were user-defined.
```

Comment:
I would like a comment that explains the difference between the two versions of build_getter_setter.